File: openapi/code_samples/java/api_users_logout.java
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/user")
public class UserController {

    @PostMapping("/logout")
    public ResponseEntity<Object> logout() {
        // Logic to log out the user and invalidate the token
        if (logoutUser()) {
            return ResponseEntity.ok(new UserDTO("User logged out successfully"));
        }

        return ResponseEntity.badRequest().body(new Fault("Logout failed"));
    }

    private boolean logoutUser() {
        // Implement your logout logic here
        // For demonstration purposes, assume the user is successfully logged out
        return true;
    }
}

class UserDTO {
    private String message;

    public UserDTO(String message) {
        this.message = message;
    }

    // getter
}

File: openapi/code_samples/csharp/api_users_logout.cs
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;

[ApiController]
[Route("api/[controller]")]
public class UserController : ControllerBase
{
    [HttpPost("logout")]
    [SwaggerOperation(Summary = "This is used to logout an existing user. This will mark also the token as unusable once called.")]
    [ProducesResponseType(typeof(UserDTO), 200)]
    [ProducesResponseType(typeof(Fault), 400)]
    public IActionResult Logout()
    {
        // Logic to log out the user and invalidate the token
        if (LogoutUser())
        {
            return Ok(new UserDTO { Message = "User logged out successfully" });
        }

        return BadRequest(new Fault { ErrorMessage = "Logout failed" });
    }

    private bool LogoutUser()
    {
        // Implement your logout logic here
        // For demonstration purposes, assume the user is successfully logged out
        return true;
    }
}

public class UserDTO
{
    public string Message { get; set; }
}
